### PR TITLE
11.0 improve shop supply

### PR DESCRIPTION
**Pull request by ddufresne**
https://isidor.numigi.net/web?#id=11095&action=292&active_id=514&model=project.task&view_type=form&menu_id=200

### CODE REVIEW

File: project_cost_report_supply/report.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from odoo.osv.expression import AND
+from odoo.tools.float_utils import float_round
+from odoo.addons.project_cost_report.report import CostReportCategory
+
+
+class ProjectCostReportWithShopSupply(models.TransientModel):
+
+    _inherit = 'project.cost.report'
+
+    def _get_empty_shop_supply_category_label(self):
```
Comment (ddufresne) [inline project_cost_report_supply/report.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report/static/src/js/report_action.js [hunk 2]
```diff
@@ -54,18 +54,18 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
...
         this.controllerURL = action.context.url;
         this.projectId = action.context.active_id || action.params.active_id;
         this.reportContext = {
             active_id: this.projectId,
-            unfolded_categories: {
-                "product": [],
-                "time": [],
-                "outsourcing": [],
-            },
+            unfolded_categories: {},
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'unfolded_categories' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report/static/src/js/report_action.js [hunk 3]
```diff
@@ -402,6 +424,10 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
 });
 
 core.action_registry.add("project_cost_report", ReportAction);
-return ReportAction;
+return {
+    ReportAction,
+    getRecordIdFromEvent,
+    getRecordNameFromEvent,
+}
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/static/src/js/report_action.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
+*/
+odoo.define("project_cost_report_supply.ReportAction", function (require) {
+"use strict";
+
+var core = require("web.core");
+var _t = core._t;
+
+var project_cost_report = require("project_cost_report");
```
Comment (ddufresne) [inline project_cost_report_supply/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'project_cost_report' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report/static/src/js/report_action.js [hunk 5]
```diff
@@ -188,11 +188,7 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
      * Fold all report categories.
      */
     fold(){
-        this.reportContext.unfolded_categories = {
-            "product": [],
-            "time": [],
-            "outsourcing": [],
-        };
+        this.getCategoryNames().forEach((c) => this.reportContext.unfolded_categories[c] = []);
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Generic Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/static/src/js/report_action.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+var ReportAction = project_cost_report.ReportAction;
+var getRecordIdFromEvent = project_cost_report.getRecordIdFromEvent;
```
Comment (ddufresne) [inline project_cost_report_supply/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'project_cost_report' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report/static/src/js/report_action.js [hunk 7]
```diff
@@ -347,6 +362,19 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
...
+     *
+     * @returns {Array} - the domain
+     */
+    getOutsourcingCategoryDomain(){
+        return [
+            ["account_id.project_ids", "=", this.projectId],
+            ["task_id", "=", false],
+            ["product_id.type", "=", "service"],
+            ["revenue", "=", false],
+        ]
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/static/src/js/report_action.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+     * @param {jQuery.Event} - the click event
+     */
+    shopSupplyTotalClicked(event){
+        event.preventDefault();
+        var categoryId = getRecordIdFromEvent(event);
+        var categoryName = getRecordNameFromEvent(event);
+        var actionName = _t("Analytic Lines ({category})").replace("{category}", categoryName);
+        this.do_action({
+            name: actionName,
+            res_model: "account.analytic.line",
```
Comment (ddufresne) [inline project_cost_report_supply/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'res_model' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report/static/src/js/report_action.js [hunk 9]
```diff
@@ -2,7 +2,7 @@
     © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
     License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
 */
-odoo.define("project_cost_report.ReportAction", function (require) {
+odoo.define("project_cost_report", function (require) {
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: ['odoo' is not defined. (no-undef)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/static/src/js/report_action.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+odoo.define("project_cost_report_supply.ReportAction", function (require) {
+"use strict";
+
+var core = require("web.core");
+var _t = core._t;
+
+var project_cost_report = require("project_cost_report");
+var ReportAction = project_cost_report.ReportAction;
+var getRecordIdFromEvent = project_cost_report.getRecordIdFromEvent;
+var getRecordNameFromEvent = project_cost_report.getRecordNameFromEvent;
```
Comment (ddufresne) [inline project_cost_report_supply/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'project_cost_report' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/static/src/js/report_action.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
+/*
+    © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
+*/
+odoo.define("project_cost_report_supply.ReportAction", function (require) {
```
Comment (ddufresne) [inline project_cost_report_supply/static/src/js/report_action.js]: Codacy Issue found: ['odoo' is not defined. (no-undef)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/static/src/js/report_action.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+var core = require("web.core");
+var _t = core._t;
+
+var project_cost_report = require("project_cost_report");
+var ReportAction = project_cost_report.ReportAction;
+var getRecordIdFromEvent = project_cost_report.getRecordIdFromEvent;
+var getRecordNameFromEvent = project_cost_report.getRecordNameFromEvent;
+
+ReportAction.include({
+    events: _.extend({}, ReportAction.prototype.events, {
```
Comment (ddufresne) [inline project_cost_report_supply/static/src/js/report_action.js]: Codacy Issue found: ['_' is not defined. (no-undef)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_wip_supply_cost/models/account_move_line.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline project_wip_supply_cost/models/account_move_line.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_wip_supply_cost/models/account_move_line.py [hunk 14]
```diff
@@ -0,0 +1,20 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline project_wip_supply_cost/models/account_move_line.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/static/src/js/report_action.js [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
+*/
+odoo.define("project_cost_report_supply.ReportAction", function (require) {
+"use strict";
+
+var core = require("web.core");
+var _t = core._t;
+
+var project_cost_report = require("project_cost_report");
+var ReportAction = project_cost_report.ReportAction;
```
Comment (ddufresne) [inline project_cost_report_supply/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'project_cost_report' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report_supply/report.py [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+        """Get the label to display on the unique shop supply category.
+
+        :rtype: str
+        """
+        return _('Shop Supply')
+
+    def _get_shop_supply_analytic_line_domain(self, project):
```
Comment (ddufresne) [inline project_cost_report_supply/report.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

File: project_cost_report/static/src/js/report_action.js [hunk 17]
```diff
@@ -312,6 +317,21 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
...
+     * @returns {Array} - the domain
+     */
+    getTimeCategoryDomain(event){
+        var taskTypeId = getRecordIdFromEvent(event);
+        return [
+            ["account_id.project_ids", "=", this.projectId],
+            ["task_id.task_type_id", "=", taskTypeId],
+            ["task_id", "!=", false],
+            ["revenue", "=", false],
+        ]
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

**Review by foutoucour — APPROVED** [review]

File: project_cost_report/static/src/js/report_action.js [hunk 18]
```diff
@@ -277,6 +273,21 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
...
+     * @returns {Array} - the domain
+     */
+    getProductCategoryDomain(event){
+        var categoryId = getRecordIdFromEvent(event);
+        return [
+            ["account_id.project_ids", "=", this.projectId],
+            ["product_id.type", "in", ["consu", "product"]],
+            ["product_id.categ_id", "=", categoryId],
+            ["revenue", "=", false],
+        ]
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4044060)

**Pull request merged**